packages/nx/src/utils/package-manager.ts

```diff
@@ -75,7 +73,10 @@ export function getPackageManagerCommand(
         addGlobal: 'pnpm add -g',
         rm: 'pnpm rm',
         exec: useExec ? 'pnpm exec' : 'pnpx',
-        run: (script: string, args: string) => `pnpm run ${script} -- ${args}`,
+        run: (script: string, args: string) =>
+          includeDoubleDashBeforeArgs
+            ? `pnpm run ${script} -- ${args}`
+            : `pnpm run ${script} ${args}`,
```

Comment:
In case anyone is looking to patch v13, I created this patch for `@nrwl/tao@13.8.0`:

```patch
diff --git a/src/shared/package-manager.js b/src/shared/package-manager.js
index b74ffe01bc7cb161ac46462d723d8f521ec5a849..2ced85a576ad95bc2d18b363529d721d471408b2 100644
--- a/src/shared/package-manager.js
+++ b/src/shared/package-manager.js
@@ -39,17 +39,25 @@ function getPackageManagerCommand(packageManager = detectPackageManager()) {
         }),
         pnpm: () => {
             const [major, minor] = getPackageManagerVersion('pnpm').split('.');
+            const majorVersion = +major
+
             let useExec = false;
-            if (+major >= 6 && +minor >= 13) {
+            if (majorVersion >= 6 && +minor >= 13) {
                 useExec = true;
             }
+
+            const includeDoubleDashBeforeArgs = majorVersion < 7;
+            
             return {
                 install: 'pnpm install --no-frozen-lockfile',
                 add: 'pnpm add',
                 addDev: 'pnpm add -D',
                 rm: 'pnpm rm',
                 exec: useExec ? 'pnpm exec' : 'pnpx',
-                run: (script, args) => `pnpm run ${script} -- ${args}`,
+                run: (script, args) =>
+                    includeDoubleDashBeforeArgs
+                        ? `pnpm run ${script} -- ${args}`
+                        : `pnpm run ${script} ${args}`,
                 list: 'pnpm ls --depth 100',
             };
         },
```